src/middleware/authorization.js

```diff
@@ -0,0 +1,24 @@
+import { isDevelopment } from '../config';
+import { handleRequest, FALLBACK_API_KEY, findDeveloper } from '../controllers/developers/utils';
+
+export default async (req, res, next) => {
+  try {
+    let { apiToken: apiKey } = handleRequest(req);
+
+    if (!apiKey && isDevelopment) {
+      apiKey = FALLBACK_API_KEY;
+    }
+
+    // Check if API key belongs to the requested developer
+    const developer = await findDeveloper(apiKey);
+
+    if (developer.length < 1) {
+      throw new Error('Invalid API Key. Check your API Key and try again', 403);
+    }
+    res.locals.developer = developer;
```

Comment:
i don't think we want to start the pattern of adding objects to the `res` Response object, instead you would want to do something like the following that saves the data on the `req` Request object
```suggestion
    req.locals.developer = developer;
```